template-compose/app/src/main/java/co/nimblehq/template/compose/di/modules/DataStoreModule.kt

```diff
@@ -0,0 +1,27 @@
+package co.nimblehq.template.compose.di.modules
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import androidx.datastore.preferences.preferencesDataStoreFile
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+private const val USER_PREFERENCES = "user_preferences"
+
+@Module
+@InstallIn(SingletonComponent::class)
+object DataStoreModule {
+
+    @Provides
+    @Singleton
+    fun providePreferencesDataStore(@ApplicationContext appContext: Context): DataStore<Preferences> {
+        return PreferenceDataStoreFactory.create(
+            produceFile = { appContext.preferencesDataStoreFile(USER_PREFERENCES) }
```

Comment:
Shall we also define `corruptionHandler` in case `migration` is needed?
```
            corruptionHandler = ReplaceFileCorruptionHandler(
                produceNewData = { emptyPreferences() }
            )
```

Reply:
@kaungkhantsoe Let's simplify it on our template. We don't need to handle the migration case. 🙏🏻

Reply:
We should add the `corruptionHandler`. I once read it in the article that it could lead to `IOException` if it got corrupted. @lydiasama 🙏